### DISCUSSION TITLE
Need to know something as a beginner to this field

### DISCUSSION THREAD

**Discussion by acharyayush** [TanStack/query#6637] [General]
I always think that using something like react query may be not so good about server load. We fetch everything every time we change something  from the server. Can we not do something like synchronize the server and client side store to minimize the load to server.

Or do we not need to think a lot about server load. Am i overthinking?😂 Just curious to know as a beginner

**Comment by kalijonn** [comment]
What do you mean by "fetch everything"? If you're talking about Query Invalidation, you can target only the queries that need to be invalidated by specifying the queryKey.

**Comment by TkDodo** [comment]
`staleTime` defaults to zero, you might want to adjust that. I have a lot of content on this topic, specifically:

- https://tkdodo.eu/blog/react-query-as-a-state-manager
- https://tkdodo.eu/blog/thinking-in-react-query
